src/libraries/System.Private.CoreLib/src/System/Numerics/Matrix4x4.Impl.cs

```diff
@@ -832,6 +832,24 @@ public static Impl CreateTranslation(float positionX, float positionY, float pos
                 return result;
             }
 
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static Impl CreateViewport(float x, float y, float width, float height, float minDepth, float maxDepth)
+            {
+                Impl result;
+
+                // 4x SIMD fields to get a lot better codegen
+                result.W = new Vector4(width, height, 0f, 0f);
+                result.W *= new Vector4(0.5f, 0.5f, 0f, 0f);
```

Comment:
Couldn't this be `result.W = new Vector4(width * 0.5f, height * 0.f, 0f, 0f);`?

Reply:
That's less efficient as it requires two separate multiplications and then an insert.

The way its done here is 1 insert and then 1 multiplication.